besu/src/test/java/org/hyperledger/besu/util/EphemeryGenesisUpdaterTest.java

```diff
@@ -0,0 +1,141 @@
+
+import org.hyperledger.besu.config.GenesisConfigFile;
+
+import java.math.BigInteger;
+import java.util.Map;
+import java.util.Optional;
+import java.util.TreeMap;
+
+import io.vertx.core.json.JsonObject;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+public class EphemeryGenesisUpdaterTest {
+  private static final int GENESIS_CONFIG_TEST_CHAINID = 39438135;
+  private static final long GENESIS_TEST_TIMESTAMP = 1720119600;
+  private static final long CURRENT_TIMESTAMP = 1712041200;
+  private static final long CURRENT_TIMESTAMP_HIGHER = 1922041200;
+  private static final long PERIOD_IN_SECONDS = 28 * 24 * 60 * 60;
+  private static final long PERIOD_SINCE_GENESIS = 3;
+
+  private static final JsonObject VALID_GENESIS_JSON =
+      (new JsonObject())
+          .put("config", (new JsonObject()).put("chainId", GENESIS_CONFIG_TEST_CHAINID))
+          .put("timestamp", GENESIS_TEST_TIMESTAMP);
+
+  private static final GenesisConfigFile INVALID_GENESIS_JSON = fromConfig("{}");
+  private static final JsonObject INVALID_GENESIS_JSON_WITHOUT_CHAINID =
+      (new JsonObject()).put("timestamp", GENESIS_TEST_TIMESTAMP);
+
+  private static final JsonObject INVALID_GENESIS_JSON_WITHOUT_TIMESTAMP =
+      new JsonObject()
+          .put("config", (new JsonObject()).put("chainId", GENESIS_CONFIG_TEST_CHAINID));
+
+  @Test
+  public void testEphemeryWhenChainIdIsAbsent() {
+    final GenesisConfigFile config =
+        GenesisConfigFile.fromConfig(INVALID_GENESIS_JSON_WITHOUT_CHAINID.toString());
+    Optional<BigInteger> chainId = config.getConfigOptions().getChainId();
+    assertThat(chainId).isNotPresent();
+  }
+
+  @Test
+  public void testShouldDefaultTimestampToZero() {
+    final GenesisConfigFile config =
+        GenesisConfigFile.fromConfig(INVALID_GENESIS_JSON_WITHOUT_TIMESTAMP.toString());
+    assertThat(config.getTimestamp()).isZero();
+  }
+
+  @Test
+  public void testEphemeryWhenGenesisJsonIsInvalid() {
+    assertThatThrownBy(INVALID_GENESIS_JSON::getDifficulty)
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessageContaining("Invalid genesis block configuration");
+  }
+
+  @Test
+  public void testEphemeryWhenGenesisJsonIsValid() {
+    final GenesisConfigFile config = GenesisConfigFile.fromConfig(VALID_GENESIS_JSON.toString());
+    assertThat(String.valueOf(config.getTimestamp()))
+        .isEqualTo(String.valueOf(GENESIS_TEST_TIMESTAMP));
+    assertThat(config.getConfigOptions().getChainId())
+        .hasValue(BigInteger.valueOf(GENESIS_CONFIG_TEST_CHAINID));
+    assertThat(String.valueOf(config.getTimestamp())).isNotNull();
+    assertThat(String.valueOf(config.getTimestamp())).isNotEmpty();
+    assertThat(String.valueOf(config.getConfigOptions().getChainId())).isNotNull();
+    assertThat(String.valueOf(config.getConfigOptions().getChainId())).isNotEmpty();
```

Comment:
you can delete lines 86 and 87 - they aren't doing anything because line 82 is already testing this value